standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -2213,9 +2213,9 @@ private void create_table_core(final RawStore ms, final CreateTableRequest req)
       }
 
       if (!TableType.VIRTUAL_VIEW.toString().equals(tbl.getTableType())) {
-        if (tbl.getSd().getLocation() == null
-            || tbl.getSd().getLocation().isEmpty()) {
-          tblPath = wh.getDefaultTablePath(db, tbl);
+        if (tbl.getSd().getLocation() == null || tbl.getSd().getLocation().isEmpty()) {
```

Comment:
I am wondering if there are some loopholes here. What happens if lock free drop is enabled and someone drops and recreated an acid table with manually given location? Shouldn't we only allow lock free drop if the table location is managed by hive (that would mean there won't be lockless reads for tables with manual location)

Reply:
i don't think we support custom location for acid managed tables, do we?

Reply:
I think we do, just tried it on a cluster, created a transactional table with a custom location and everything works as expected (insert, read, compaction...)